### PR TITLE
device identity

### PR DESCRIPTION
**Pull request by piotrbartman**
implements QubesOS/qubes-issues/issues/9325

### CODE REVIEW

**Comment by piotrbartman** [comment]
only dependency (`qubesadmin`) related errors left

File: qui/devices/backend.py [hunk 1]
```diff
@@ -231,8 +233,9 @@ def attach_to_vm(self, vm: VM):
         Perform attachment to provided VM.
         """
         try:
-            assignment = qubesadmin.device_protocol.DeviceAssignment(
-                self.backend_domain, self.id_string)
+            assignment = DeviceAssignment(VirtualDevice(Port(
```
Comment (marmarta) [inline qui/devices/backend.py:236]: I think this needs a bit better documentation - this is a pretty opaque way of making a device assignment and a comment would be nice here :)

File: qui/devices/device_widget.py [hunk 2]
```diff
@@ -124,7 +124,8 @@ def device_list_update(self, vm, _event, **_kwargs):
         try:
             for devclass in DEV_TYPES:
                 for device in vm.devices[devclass]:
-                    changed_devices[str(device)] = backend.Device(device, self)
+                    changed_devices[str(device.port)] = backend.Device(
```
Comment (marmarta) [inline qui/devices/device_widget.py:127]: if the variable changed_devices is now a dict that maps port to device, that should be reflected in the name or at least in comments

File: qui/devices/device_widget.py [hunk 3]
```diff
@@ -172,7 +173,8 @@ def initialize_dev_data(self):
             for devclass in DEV_TYPES:
                 try:
                     for device in domain.devices[devclass]:
-                        self.devices[str(device)] = backend.Device(device, self)
+                        self.devices[str(device.port)] = backend.Device(
```
Comment (marmarta) [inline qui/devices/device_widget.py:176]: same here - it's very non-obvious that devices are indexed by port

**Review by marmarta — CHANGES_REQUESTED** [review]
I will also today test it on a test laptop for a more meaningful comments on the UI

**Comment by marmarta** [comment]
So, I found a double bug. To reproduce: 
- install all packages relevant to this
- reboot
- make a device
- `qvm-device block assign --ask d12test f40xfce:loop0`
- try to start d12test

Result:
- I wait endlessly, nothing is happening (I think the daemon to show question failed to start?)
- worse, everything that uses qubes events hangs (qui-*, qvm-ls, everything) - something that should be async is not async?

**Review by marmarek — APPROVED** [review]

**Pull request merged**